### PR TITLE
feat: slpk from local filesystem feature

### PR DESCRIPTION
**Pull request by dariaterekhova-actionengine**
First part of changes for slpk from local filesystem feature

### CODE REVIEW

**Comment by dsavinov-actionengine** [comment]
@belom88 Please review as soon as you can

File: src/redux/slices/flattened-sublayers-slice.ts [hunk 1]
```diff
@@ -144,27 +147,43 @@ const getLayersAndSublayers = async (
...
 }> => {
+  const options: LoaderOptions = {};
+  if (typeof tilesetData.url !== "string") {
+    try {
+      const fileProvider = await FileProvider.create(new BlobFile(tilesetData.url));
+      const archive = await parseSLPKArchive(fileProvider, undefined, tilesetData.url.name);
+      const fileSystem = new ZipFileSystem(archive);
+      options.fetch = fileSystem.fetch.bind(fileSystem) as (filename: string) => Promise<Response>;
+    } catch (e) {
+      console.log("error", e);
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.ts:165]:
```suggestion
      console.error(e);
```

File: src/redux/slices/flattened-sublayers-slice.ts [hunk 2]
```diff
@@ -144,27 +147,43 @@ const getLayersAndSublayers = async (
...
-    url: string;
+    url: string | File;
     visibility: boolean;
     token?: string;
     type: TilesetType | undefined;
   }>;
   sublayers: Sublayer[];
 }> => {
+  const options: LoaderOptions = {};
+  if (typeof tilesetData.url !== "string") {
```
Comment (belom88) [inline src/redux/slices/flattened-sublayers-slice.ts:158]: Add a test case

File: src/pages/viewer-app/viewer-app.tsx [hunk 3]
```diff
@@ -238,8 +248,9 @@ export const ViewerApp = () => {
     setSelectedFeatureAttributes(null);
     setSelectedFeatureIndex(-1);
     if (buildingExplorerOpened && tilesetsData[0]) {
+      const tilesetDataUrl = typeof tilesetsData[0].url === "string" ? tilesetsData[0].url : tilesetsData[0].url.name;
```
Comment (belom88) [inline src/pages/viewer-app/viewer-app.tsx:251]: Some utility `function getDatasetUrl(dataset: LayerExample | TilesetMetadata) => string`. You reuse this pattern many times. This function would make the code more readable.

```suggestion
      const tilesetDataUrl = getDatasetUrl(tilesetsData[0]);
```

File: src/pages/viewer-app/viewer-app.tsx [hunk 4]
```diff
@@ -216,7 +220,13 @@ export const ViewerApp = () => {
     const tilesetsData: TilesetMetadata[] = [];
 
     for (const layer of activeLayers) {
-      const params = parseTilesetUrlParams(layer.url, layer);
+      let params = { tilesetUrl: "" as string | File, token: "" };
+      if (typeof layer.url === "string") {
+        params = parseTilesetUrlParams(layer.url, layer);
+      } else {
+        params.tilesetUrl = layer.url;
+      }
```
Comment (belom88) [inline src/pages/viewer-app/viewer-app.tsx:228]: Why this part is different in `src/pages/debug-app/debug-app.tsx`  and `src/components/comparison/comparison-side/comparison-side.tsx`?

**Review by belom88 — CHANGES_REQUESTED** [review]

File: src/utils/layer-utils.ts [hunk 5]
```diff
@@ -95,7 +95,7 @@ export const findExampleAndUpdateWithViewState = (
 
   for (const example of examplesCopy) {
     // We can't compare by tileset.url === example.url because BSL and Scene examples url is not loaded as tileset.
-    if (tileset.url.includes(example.url) && !example.viewState) {
+    if (tileset.url.includes(typeof example.url === "string" ? example.url : example.url.name) && !example.viewState) {
```
Comment (belom88) [inline src/utils/layer-utils.ts:98]:
```suggestion
    if (tileset.url.includes(getLayerUrl(example.url)) && !example.viewState) {
```

**Review by belom88 — APPROVED** [review]

**Pull request merged**